staging/src/k8s.io/client-go/discovery/fake/discovery_test.go

```diff
@@ -44,3 +47,19 @@ func TestFakingServerVersion(t *testing.T) {
 		t.Fatalf("unexpected faked discovery return value: %q", sv.GitCommit)
 	}
 }
+
+func TestFakingServerVersionWithError(t *testing.T) {
+	expectedError := errors.New("an error occurred")
+	fakeClient := fakeclientset.NewSimpleClientset()
+	fakeClient.Discovery().(*fakediscovery.FakeDiscovery).PrependReactor("*", "*", func(action kubetesting.Action) (handled bool, ret runtime.Object, err error) {
+		return true, nil, expectedError
+	})
+
+	_, err := fakeClient.Discovery().ServerVersion()
+	if err == nil {
+		t.Fatal("ServerVersion should return error, returned nil instead")
+	}
+	if err.Error() != expectedError.Error() {
```

Comment:
Direct comparison should work here (and if it does not, `errors.Is` is better than string comparison):

```suggestion
	if err != expectedError {
```

Reply:
Confirmed, direct comparison does work.